lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -21,11 +46,16 @@ def initialize(serializer, options = {})
         def serializable_hash(options = nil)
           options ||= {}
 
-          if serializer.respond_to?(:each)
-            serializable_hash_for_collection(options)
-          else
-            serializable_hash_for_single_resource(options)
-          end
+          hash =
+            if serializer.respond_to?(:each)
+              serializable_hash_for_collection(options)
+            else
+              serializable_hash_for_single_resource(options)
+            end
+
+          ApiObjects::JsonApi.include_member? && hash[:jsonapi] = ApiObjects::JsonApi.member
```

Comment:
that is, I was thinking that each ApiObject would know how to build the json api object it represents